src/stack/phy/ChannelModel/LteRealisticChannelModel.cc

```diff
@@ -2156,38 +2152,63 @@ bool LteRealisticChannelModel::error_Mode4(LteAirFrame *frame, UserControlInfo*
 
             //Get the Bler
             averageSnr += dBToLinear(snrV[jt->first]);
+            averageSinr += dBToLinear(sinrVector[jt->first]);
             countUsedRbs += 1;
         }
     }
 
     averageSnr = linearToDb(averageSnr/countUsedRbs);
+    averageSinr = linearToDb(averageSinr/countUsedRbs);
 
     if (averageSnr > binder_->phyPisaData.maxSnr())
-        bler = 0;
+        blerSnr = 0;
     else
     if (lteInfo->getFrameType() == SCIPKT)
     {
         // TODO: Make this slightly tidier.
-        bler = binder_->phyPisaData.GetPscchBler(binder_->phyPisaData.AWGN, binder_->phyPisaData.SISO, averageSnr);
+        blerSnr = binder_->phyPisaData.GetPscchBler(binder_->phyPisaData.AWGN, binder_->phyPisaData.SISO, averageSnr);
     }
     else
     {
         if (analytical_)
-            bler = binder_->phyPisaData.GetBlerAnalytical(mcs, averageSnr);
+            blerSnr = binder_->phyPisaData.GetBlerAnalytical(mcs, averageSnr);
         else
-            bler = binder_->phyPisaData.GetPsschBler(binder_->phyPisaData.AWGN, binder_->phyPisaData.SISO, mcs, averageSnr);
+            blerSnr = binder_->phyPisaData.GetPsschBler(binder_->phyPisaData.AWGN, binder_->phyPisaData.SISO, mcs, averageSnr);
+    }
+
+    if (averageSinr > binder_->phyPisaData.maxSnr())
+        blerSinr = 0;
+    else
+    if (lteInfo->getFrameType() == SCIPKT)
+    {
+        // TODO: Make this slightly tidier.
+        blerSinr = binder_->phyPisaData.GetPscchBler(binder_->phyPisaData.AWGN, binder_->phyPisaData.SISO, averageSinr);
+    }
+    else
+    {
+        if (analytical_)
+            blerSinr = binder_->phyPisaData.GetBlerAnalytical(mcs, averageSinr);
+        else
+            blerSinr = binder_->phyPisaData.GetPsschBler(binder_->phyPisaData.AWGN, binder_->phyPisaData.SISO, mcs, averageSinr);
     }
 
     double er = uniform(getEnvir()->getRNG(0),0.0, 1.0);
 
-    if (er <= bler)
+    bool resultSnr = true;
+    bool resultSinr = true;
+
+    if (er <= blerSnr)
     {
         // Signal too weak, we can't receive it
-        return false;
+        resultSnr = false;
     }
-    // Signal is strong enough, receive this Signal
 
-    return true;
+    if (er <= blerSnr) {
```

Comment:
I think the comparison should be `er <= blerSinr`

Reply:
I agree, thanks for pointing out my mistake. Unfortunately I'm out of the office today so I wont be able to make the change today.

Reply:
You're welcome, Samuel. Please, let me know how the results look like with this modification. I'd we interested to discuss this further since I'm using the model as well.

Reply:
Hi Andres, the results I present above are for two vehicles, one transmitter and one receiver, so there is no interference present. As such, there will be no difference in the results.